src/components/RichTextEditor/Plugins/LinkPlugin/FloatingLink/InsertLinkModal/useFloatingLinkInsert.ts

```diff
@@ -0,0 +1,82 @@
+import { useEffect } from 'react';
+import {
+    ELEMENT_LINK,
+    FloatingLinkProps,
+    HTMLPropsAs,
+    LinkPlugin,
+    floatingLinkActions,
+    floatingLinkSelectors,
+    focusEditor,
+    getPluginOptions,
+    triggerFloatingLinkInsert,
+    useComposedRef,
+    useEditorRef,
+    useFloatingLinkEscape,
+    useFloatingLinkSelectors,
+    useHotkeys,
+    useOnClickOutside,
+    useVirtualFloatingLink,
+} from '@udecode/plate';
+import { getSelectionBoundingClientRect } from '@udecode/plate-floating';
+import { useFocused } from 'slate-react';
+
+export const useFloatingLinkInsert = ({ floatingOptions, ...props }: FloatingLinkProps): HTMLPropsAs<'div'> => {
+    const editor = useEditorRef();
+    const focused = useFocused();
+    const mode = useFloatingLinkSelectors().mode();
+    const open = useFloatingLinkSelectors().isOpen(editor.id);
+
+    const { triggerFloatingLinkHotkeys } = getPluginOptions<LinkPlugin>(editor, ELEMENT_LINK);
+
+    useHotkeys(
+        triggerFloatingLinkHotkeys!,
+        (e) => {
+            e.preventDefault();
+
+            triggerFloatingLinkInsert(editor, {
+                focused,
+            });
+        },
+        {
+            enableOnContentEditable: true,
+        },
+        [focused],
+    );
+
+    const ref = useOnClickOutside(() => {
```

Comment:
👍 you probably need the same thing for the edit link, since when i open the edit modal, it doesn't close the dialog when editing a link in another editor